File: test/config_test.erl
-module(config_test).
-compile([export_all]).
-include_lib("eunit/include/eunit.hrl").
-include("src/geef_records.hrl").

config_test_() ->
    {foreach, fun start/0, fun stop/1, [fun bool_test/1,
					fun string_test/1]}.

start() ->
    {A, B, C} = now(),
    N = node(),
    TmpFile = io_lib:format("/tmp/geef-~p~p~p~p.gitconfig", [N, A, B, C]),
    {ok, Config} = geef_config:open(TmpFile),
    {Config, TmpFile}.

bool_test({Config, _}) ->
    Var = "core.logallrefupdates",
    ok = geef_config:set(Config, Var, true),
    [?_assertEqual(ok, geef_config:set(Config, Var, true)),
     ?_assertEqual({ok, true}, geef_config:get_bool(Config, Var))].

string_test({Config, _}) ->
    Var = "user.name",
    Val = <<"Random J. Hacker">>,
    [?_assertEqual(ok, geef_config:set(Config, Var, Val)),
     ?_assertEqual({ok, Val}, geef_config:get_string(Config, Var))].

stop({_, Path}) ->
    ok = file:delete(Path).

File: test/config_test.exs
defmodule ConfigTest do
  use ExUnit.Case
  use Geef
  import RepoHelpers


  setup do
    {repo, path} = tmp_bare
    Process.link(repo)
    {:ok, config} = Repository.config(repo)
    on_exit(fn -> File.rm_rf!(path) end)
    {:ok, [config: config, path: path]}
  end

  test "write and get a boolean", meta do
    config = meta[:config]
    var = "core.logallrefupdates"
    assert :ok = Config.set(config, var, true)
    assert {:ok, true} = Config.get_bool(config, var)
  end

  test "write and get a string", meta do
    config = meta[:config]
    var = "user.name"
    val = "Random J. Hacker"
    assert :ok = Config.set(config, var, val)
    assert {:ok, ^val} = Config.get_string(config, var)
  end

end
